File: app/src/main/java/com/tejasbhitle/gallery/enums/Sort.java
package com.tejasbhitle.gallery.enums;

public enum Sort {
    DATE_ASCEND("DATE_ASCEND"),
    DATE_DESCEND("DATE_DESCEND"),
    NAME_ASCEND("NAME_ASCEND"),
    NAME_DESCEND("NAME_DESCEND");

    private String sort;
    Sort(String sort){
        this.sort = sort;
    }

    @Override
    public String toString() {
        return sort;
    }

    public static Sort getSort(String sort){
        if(sort.equals(DATE_ASCEND.toString()))
            return DATE_ASCEND;
        if(sort.equals(DATE_DESCEND.toString()))
            return DATE_DESCEND;
        if(sort.equals(NAME_ASCEND.toString()))
            return NAME_ASCEND;
        if(sort.equals(NAME_DESCEND.toString()))
            return NAME_DESCEND;
        return NAME_ASCEND;
    }
}

File: app/src/main/java/com/tejasbhitle/gallery/enums/Sort.kt
package com.tejasbhitle.gallery.enums

enum class Sort private constructor(private val sort: String) {
    DATE_ASCEND("DATE_ASCEND"),
    DATE_DESCEND("DATE_DESCEND"),
    NAME_ASCEND("NAME_ASCEND"),
    NAME_DESCEND("NAME_DESCEND");

    override fun toString(): String {
        return sort
    }

    companion object {

        fun getSort(sort: String): Sort {
            return when(sort){
                DATE_ASCEND.toString() ->  DATE_ASCEND
                DATE_DESCEND.toString() -> DATE_DESCEND
                NAME_DESCEND.toString() -> NAME_DESCEND
                else -> return NAME_ASCEND
            }
        }
    }
}
